esi_leap/api/controllers/v1/contract.py

```diff
@@ -137,10 +138,36 @@ def delete(self, contract_id):
 
         permitted.cancel()
 
+    @staticmethod
+    def _get_offer_by_uuid_or_name(uuid_or_name):
+        if uuidutils.is_uuid_like(uuid_or_name):
+            o = offer.Offer.get(uuid_or_name)
+            if o.status != statuses.AVAILABLE:
+                return []
+            else:
+                return [o]
+        else:
+            return offer.Offer.get_all({'name': uuid_or_name,
+                                        'status': statuses.AVAILABLE})
```

Comment:
In the vein of ensuring that `get` functions only return one thing, I think it makes sense to expand these functions a bit:

* move any needed permissions checking here
* move any exception handling here